esmvaltool/diag_scripts/shared/_base.py

```diff
@@ -516,13 +516,9 @@ def main(cfg):
         'log.txt', 'profile.bin', 'resource_usage.txt', 'settings.yml'
     }
 
-    output_directories = []
-    if cfg['write_netcdf']:
-        output_directories.append(cfg['work_dir'])
-    if cfg['write_plots']:
-        output_directories.append(cfg['plot_dir'])
-
-    old_content = [p for p in output_directories if os.path.exists(p)]
+    output_directories = (cfg['work_dir'], cfg['plot_dir'])
+    old_content = [p for p in output_directories
+                   if Path(p).exists() and any(Path(p).iterdir())]
```

Comment:
Due to the addition of `any(Path(p).iterdir())` the re-run is possible when `work_dir` or `plot_dir` is empty, but this fails later with

```
FileExistsError: [Errno 17] File exists: '/scratch/b/b309141/work_v2/recipe_python_20210216_113224/plots/timeseries/script1'
```

due to

https://github.com/ESMValGroup/ESMValTool/blob/bb4a9cf4da2cc0be14c029c669716e7002e261ea/esmvaltool/diag_scripts/shared/_base.py#L546

Can remove `any(Path(p).iterdir())`? I think this is the safest option here.

Reply:
It seems to work fine for me, what commands are you using to run the diagnostic? I tried `esmvaltool run examples/recipe_python.yml` and then re-running with the command displayed while running, that fails as expected without the `-i` or `-f` option and seems to work fine with these options.

Reply:
I didn't add any option because I think when the directories do not exist it should work without any option (it does at least with the current `master´).

Reply:
When I re-run the diagnostic script without any option, i.e.:
```
python /home/bandela/src/esmvalgroup/esmvaltool/esmvaltool/diag_scripts/examples/diagnostic.py /home/bandela/esmvaltool_output/recipe_python_20210216_114358/run/map/script1/settings.yml
```
I get the following error message:
```
FileExistsError: Script will abort to prevent accidentally overwriting your data in the following output files or directories:
/home/bandela/esmvaltool_output/recipe_python_20210216_114358/work/map/script1
/home/bandela/esmvaltool_output/recipe_python_20210216_114358/plots/map/script1
/home/bandela/esmvaltool_output/recipe_python_20210216_114358/run/map/script1/diagnostic_provenance.yml
 Use -f or --force to force emptying the output directories or use -i or --ignore-existing to ignore existing output directories.
```
which looks OK to me.

What commands are you running exactly to produce the error you're seeing here https://github.com/ESMValGroup/ESMValTool/pull/2018#discussion_r576758750?

Reply:
Yes, when immediately re-running I get the same error message, but after deleting `/home/bandela/esmvaltool_output/recipe_python_20210216_114358/work/map/script1`, `/home/bandela/esmvaltool_output/recipe_python_20210216_114358/run/map/script1/diagnostic_provenance.yml` and the **contents** of `/home/bandela/esmvaltool_output/recipe_python_20210216_114358/plots/map/script1`, I get the following errors:

In current `master`:
```
FileExistsError: Script will abort to prevent accidentally overwriting your data in the following output files or directories:
/scratch/b/b309141/work_v2/recipe_python_20210216_113224/work/timeseries/script1
 Use -f or --force to force emptying the output directories or use -i or --ignore-existing to ignore existing output directories.
```
(which is what I expect)

In this branch:
```
FileExistsError: [Errno 17] File exists: '/scratch/b/b309141/work_v2/recipe_python_20210216_113224/work/timeseries/script1'
```
(which is what I do not expect).

The reason it this addition: `any(Path(p).iterdir())`

Reply:
Thanks for clarifying, I hadn't considered the possibility that people would manually delete stuff. I thought it might be a bit nicer to not complain if there were just empty output directories, that's why I added the `any(Path(p).iterdir())`. Could you try again with the latest changes, please?

Reply:
Tested succesfully